src/back/glsl/mod.rs

```diff
@@ -827,6 +827,11 @@ impl<'a, W: Write> Writer<'a, W> {
 
             let fun_info = &self.info[handle];
 
+            // Skip functions that that are not compatible with this entry point's stage
```

Comment:
Here and elsewhere:

```suggestion
            // Skip functions that that are not compatible with this entry point's stage.
            // When validation is enabled, we will reject programs that call this, so
            // this will only leave calls dangling when validation is disabled.
```

Reply:
We will reject programs that call this but even with validation enabled, we will still write the function declaration.

Reply:
Okay - how about this:

> When validation is disabled, `fun_info.available_stages` is always just `ShaderStages::all()`, so this will write all functions in the module, and the downstream GLSL compiler will catch any problems. When validation is enabled, it rejects modules whose entry points try to call incompatible functions, so if we got this far, then any functions incompatible with our selected entry point must not be used.